connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -0,0 +1,84 @@
+import type { Client } from "@microsoft/microsoft-graph-client";
+import type * as MicrosoftGraph from "@microsoft/microsoft-graph-types";
+
+export async function getSites(client: Client) {
+  const res = await client
+    .api("/sites?search=*")
+    .select("id,name,displayName")
+    .get();
+  return res.value as MicrosoftGraph.Site[];
+}
+
+export async function getDrives(client: Client, siteId: string) {
+  const res = await client
+    .api(`/sites/${siteId}/drives`)
+    .select("id,name")
+    .get();
+  return res.value as MicrosoftGraph.Drive[];
+}
+
+export async function getFilesAndFolders(
+  client: Client,
+  driveId: string,
+  parentId?: string
+) {
+  const parent = parentId ? `items/${parentId}` : "root";
+  const res = await client
+    .api(`/drives/${driveId}/${parent}/children`)
+    .select("id,displayName")
+    .get();
+  return res.value as MicrosoftGraph.DriveItem[];
+}
+
+export async function getFolders(
+  client: Client,
+  driveId: string,
+  parentId?: string
+) {
+  const res = await getFilesAndFolders(client, driveId, parentId);
+  return res.filter((item) => item.folder);
+}
+
+export async function getTeams(client: Client) {
+  const res = await client
+    .api("/me/joinedTeams")
+    .select("id,displayName")
+    .get();
+  return res.value as MicrosoftGraph.Team[];
+}
+
+export async function getChannels(client: Client, teamId: string) {
+  const res = await client
+    .api(`/teams/${teamId}/channels`)
+    .select("id,displayName")
+    .get();
+  return res.value as MicrosoftGraph.Channel[];
+}
+
+export async function getMessages(
+  client: Client,
+  teamId: string,
+  channelId: string
+) {
+  const res = await client
+    .api(`/teams/${teamId}/channels/${channelId}/messages`)
+    .get();
+  return res.value as MicrosoftGraph.Message[];
+}
+
+export async function registerWebhook(client: Client, resource: string) {
```

Comment:
did we end up relying on webhooks? would be glad to get briefed IRL

Reply:
I need to check if delta are usable for channels. I was not able to make them work properly but according to the doc it should be supported, so no need to use webhooks.
For files we don't need webhooks.